src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readFile(fromFileName);
+        String[] result = createResult(data);
+        writeFile(toFileName, result);
+    }
+
+    private String[] readFile(String fromFileName) {
+        StringBuilder data = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                data.append(line).append("\n");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file", e);
+        }
+        return data.toString().split("\n");
```

Comment:
Instead of using a StringBuilder and splitting by '\n', consider reading the file line by line and processing each line as you read it. This will make the code cleaner and avoid unnecessary use of splitting, which can be less efficient.